src/pykeen/models/unimodal/quate.py

```diff
@@ -0,0 +1,245 @@
+    r"""An implementation of QuatE [zhang2019]_.
+
+    QuatE uses hypercomplex valued representations for the
+    entities and relations. Entities and relations are represented as vectors
+    $\textbf{e}_i, \textbf{r}_i \in \mathbb{H}^d$, and the plausibility score is computed using the
+    quaternion inner product.
+
+    .. seealso ::
+
+        Official implementation: https://github.com/cheungdaven/QuatE/blob/master/models/QuatE.py
+    ---
+    citation:
+        author: Zhang
+        year: 2019
+        link: https://arxiv.org/abs/1904.10281
+        github: cheungdaven/quate
+    """
+
+    #: The default strategy for optimizing the model's hyper-parameters
+    hpo_default: ClassVar[Mapping[str, Any]] = dict(
+        embedding_dim=DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE,
+    )
+    #: The default loss function class
+    loss_default: ClassVar[Type[Loss]] = BCEWithLogitsLoss
+    #: The default parameters for the default loss function class
+    loss_default_kwargs: ClassVar[Mapping[str, Any]] = dict(reduction='mean')
+    #: The regularizer used by [zhang2019]_ for QuatE.
+    regularizer_default: ClassVar[Type[Regularizer]] = LpRegularizer
+    #: The LP settings used by [zhang2019]_ for QuatE.
+    regularizer_default_kwargs: ClassVar[Mapping[str, Any]] = dict(
+        weight=0.0025,
+        p=3.0,
+        normalize=True,
+    )
+
+    def __init__(
+        self,
+        triples_factory: TriplesFactory,
+        embedding_dim: int = 100,
+        normalize_relations: bool = True,
+        loss: Optional[Loss] = None,
+        regularizer: Optional[Regularizer] = None,
+        preferred_device: DeviceHint = None,
+        random_seed: Optional[int] = None,
+        entity_initializer: Hint[Initializer] = init_quaternions,
+        relation_initializer: Hint[Initializer] = init_quaternions,
+    ) -> None:
+        """Initialize QuatE.
+
+        :param triples_factory:
+            The triple factory connected to the model.
+        :param embedding_dim:
+            The embedding dimensionality of the entity embeddings.
+        :param loss:
+            The loss to use. Defaults to BCEWithLogitsLoss.
+        :param regularizer:
+            The regularizer to use.
+        :param preferred_device:
+            The default device where to model is located.
+        :param random_seed:
+            An optional random seed to set before the initialization of weights.
+        """
+        super().__init__(
+            triples_factory=triples_factory,
+            loss=loss,
+            preferred_device=preferred_device,
+            random_seed=random_seed,
+            regularizer=regularizer,
+            entity_representations=EmbeddingSpecification(
+                embedding_dim=4 * embedding_dim,
+                initializer=entity_initializer,
+                initializer_kwargs=dict(num_elements=self.num_entities, dim=embedding_dim),
+                dtype=torch.float,
```

Comment:
@mberr should we add something tricky in the dtype to flag that it's quaternions so we don't have to do this 4* thing everywhere?

@mali-git let's try and make the packing/unpacking of quaternions look like how it does for ComplEx

Reply:
> @mberr should we add something tricky in the dtype to flag that it's quaternions so we don't have to do this 4* thing everywhere?

Since torch itself does not provide such, I would wait until we have at least one additional model using quaternions. For complex we mainly did this since we wanted to make transitioning to the built-in complex numbers easier.